src/main/java/edu/umich/its/cpm/EmailFormatter.java

```diff
@@ -0,0 +1,161 @@
+package edu.umich.its.cpm;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+
+import javax.activation.DataHandler;
+import javax.mail.BodyPart;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Session;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.util.ByteArrayDataSource;
+import javax.servlet.http.HttpServletRequest;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Created by pushyami on 8/17/16.
+ */
+public class EmailFormatter {
+    private static final Logger log = LoggerFactory.getLogger(EmailFormatter.class);
+    public AttachmentHandler attachmentHandler;
+    public HttpServletRequest request;
+
+    private String emailMessage;
+
+    @Autowired
+    private Environment env;
+
+    Map<String, Object> messageMap = null;
+
+    public EmailFormatter(String emailMgs, AttachmentHandler attachmentHandler, HttpServletRequest request) throws IOException {
```

Comment:
Rather than pass the HttpServletRequest through EmailFormatting just add it directly to the Attachment Handler class when the class is created.  It is not a problem to make a new class instance whenever the request would be different.